gomponents.go

```diff
@@ -254,9 +254,47 @@ func Group(children []Node) Node {
 
 // If condition is true, return the given Node. Otherwise, return nil.
 // This helper function is good for inlining elements conditionally.
+// This doesn't behave entirely like a regular if statement, because nested Nodes and their parameters are evaluated
+// from the deepest level and up before hitting If (because that's how Go works).
+// If you need lazy evaluation of the given Node, see Lazy.
 func If(condition bool, n Node) Node {
 	if condition {
 		return n
 	}
 	return nil
 }
+
+// Lazy returns a Node which calls the given callback function only when rendered.
+// Used with If, it enables lazy evaluation of the node to return, depending on the condition passed to If.
+// It defaults to being an ElementType, but can be set to AttributeType by passing a NodeType.
+// Even though nodeType is variadic, only the first NodeType is used.
+func Lazy(cb func() Node, nodeType ...NodeType) Node {
```

Comment:
Two thoughts:
1. This seems to be complicated by implementing the optional nodeTypeDescriber interface, but I've been reading through the code and it seems like the only use of nodeTypeDescriber is to omit the output if the type is wrong here https://github.com/maragudk/gomponents/blob/1f0bb2a2aaecc7813b9eda1c0579434e9ad5fe56/gomponents.go#L110 so maybe Lazy can just not implement it without much being lost
2. If NodeType is important, and there are only two node types, maybe create a LazyAttr function and a LazyEl function?

Reply:
@vikstrous2 thank you for the review! Sorry for the delay in replying, the winter and its various sicknesses has not been kind to me and my family this year… 🤒

Yes, `nodeTypeDescriber` is only used for checking whether a node should be rendered as an element or attribute. I think it's pretty useful in e.g. form checkboxes and the `checked` attribute, and it feels wrong to leave out support for attributes, even though it would simplify the API quite a bit here.

`LazyAttr` would also solve this, arguably more elegantly, but I would really like there to not be two functions for this already slightly edge-case functionality.

Hmm. I'm seriously considering dropping this whole `Lazy` stuff, it doesn't quite feel right.